metaserver/src/test/scala/tests/search/SymbolIndexTest.scala

```diff
@@ -51,28 +52,70 @@ object SymbolIndexTest extends MegaSuite {
         line: Int,
         column: Int,
         expected: String
-    ): Unit = {
-      val term = indexer.findSymbol(path, line, column)
+    ): Option[Symbol] = {
+      val symbol = indexer.findSymbol(path, line, column)
+      Predef.assert(
+        symbol.isDefined,
+        s"Symbol not found at $path:$line:$column."
+      )
+      assertNoDiff(symbol.get.syntax, expected)
       Predef.assert(
-        term.isDefined,
-        s"Symbol not found at $path:$line:$column. Did you run scalametaEnableCompletions from sbt?"
+        indexer.symbolIndexer.get(symbol.get).isDefined,
+        s"Symbol ${symbol} is not found in the index. Did you run scalametaEnableCompletions from sbt?"
       )
-      assertNoDiff(term.get.symbol, expected)
+      symbol
+    }
+
+    def assertSymbolDefinition(
+        expected: String
+    )(
+        symbol: Symbol
+    ): Unit = {
+      val data = indexer.definitionData(symbol)
       Predef.assert(
-        term.get.definition.isDefined,
-        s"Definition not found for term $term"
+        data.isDefined,
+        s"Definition not found for term ${symbol}"
       )
+      assertNoDiff(data.get.symbol, expected)
     }
 
-    "fallback" - {
+    "find symbol and its definition" - {
       "<<User>>(...)" -
-        assertSymbolFound(3, 17, "_root_.a.User#")
+        assertSymbolFound(3, 17, "_root_.a.User.").map(
+          assertSymbolDefinition("_root_.a.User#")
+        )
       "User.<<apply>>(...)" -
-        assertSymbolFound(3, 22, "_root_.a.User#")
+        assertSymbolFound(
+          3,
+          22,
+          "_root_.a.User.apply(Ljava/lang/String;I)La/User;."
+        ).map(
+          assertSymbolDefinition("_root_.a.User#")
+        )
+      "User.<<copy>>(...)" -
+        assertSymbolFound(
```

Comment:
can we create a `assertDefinition` with 4 params that does both symbolFound and definition? I prefer to keep logic to a single check/assertion statement in every unit test

Reply:
OK. I just thought you may want to test these things separately. I'll make it into one `assertDefinition`. What about checking result definition location, does it worth checking or the symbol name is enough?

Reply:
We can keep the current methods, just add one more that does both assertSymbolFound and assertSymbolDefinition to avoid the duplication in every unit test.

Reply:
One utest question here: what is the proper way of failing in tests? Is it just throwing exceptions?

I'd like to rewrite 

```scala
val foo = somethingReturningOption()
Predef.assert(
  foo.isDefined,
  s"foo is not defined"
)
```

to 

```scala
val foo = somethingReturningOption().getOrElse(
  fail(s"foo is not defined")
)
```

So that I don't need to write `foo.get` in the following code.

Reply:
add a `fail(msg: String): Nothing` method in MegaSuite and throw exception with empty stack trace

Reply:
the method already exists it seems!

Reply:
Oh, right! I missed it.

Reply:
Addressed in 3548ab933e71621b13af599261d5da2f16e65567